src/main/kotlin/com/cultureamp/eventsourcing/PGSessionAdvisoryBookmarkLock.kt

```diff
@@ -0,0 +1,70 @@
+package com.cultureamp.eventsourcing
+
+import org.slf4j.LoggerFactory
+import java.sql.Connection
+import java.sql.SQLException
+import java.util.UUID
+
+private val logger = LoggerFactory.getLogger(PGSessionAdvisoryBookmarkLock::class.java)
+class PGSessionAdvisoryBookmarkLock(
+    private val lockTimeoutMs: Int = 60_000,
```

Comment:
Interesting - what does the timeout look like in the ruby event framework?

Reply:
[They sleep for 1 unspecified unit](https://github.com/cultureamp/event_framework/blob/master/lib/event_framework/event_processor_supervisor.rb#L50C19-L50C48) (second?) They use `try_advisory_lock` which returns immediately regardless of if you have the lock or not. I'm open to that approach but it makes things a bit more complex, so I'd be looking for a reason to do it

Reply:
oh crap sorry, so this is the timeout to acquire the lock, not the timeout for _keeping_ the lock? As in "try to get this for 60sec" vs "get this, but then let go of it after 60sec"

Reply:
yeah it's timeout waiting for the lock to become available. if we don't get it within that time some exception is thrown

Reply:
We keep it until we either return it or close the connection